### PR TITLE
[hypo71] Use USE_TRIAL_POSITION rather than DISABLE_LAST_LOC in examp…

### PR DESCRIPTION
**Pull request by gempa-jabe**
…le profiles and set USE_TRIAL_POSITION to false by default as discussed in #149.

### CODE REVIEW

**Comment by jmsaurel** [comment]
Sorry, commented on the wrong place

**Comment by jmsaurel** [comment]
Ok for the two profile files modification

File: src/ipgp/plugins/locator/hypo71/hypo71.cpp [hunk 1]
```diff
@@ -116,7 +116,7 @@ Hypo71::Hypo71() {
 		_allowedParameters.push_back("XFAR");
 		_allowedParameters.push_back("POS");
 		_allowedParameters.push_back("KAZ");
-		_allowedParameters.push_back("DISABLE_LAST_LOC");
+		_allowedParameters.push_back("USE_TRIAL_POSITION");
```
Comment (jmsaurel) [inline src/ipgp/plugins/locator/hypo71/hypo71.cpp]: Ok

File: src/ipgp/plugins/locator/hypo71/hypo71.cpp [hunk 2]
```diff
@@ -3332,7 +3332,7 @@ void Hypo71::updateProfile(const string& name) {
 		setParameter("CRUSTAL_DEPTH_MODEL", cdm);
 
 		// Control card parameters
-		string ztr, xnear, xfar, pos, kms, kfm, imag, kaz, disableLastLoc;
+		string ztr, xnear, xfar, pos, kms, kfm, imag, kaz, useTrialPosition;
```
Comment (jmsaurel) [inline src/ipgp/plugins/locator/hypo71/hypo71.cpp]: Ok

File: src/ipgp/plugins/locator/hypo71/hypo71.cpp [hunk 3]
```diff
@@ -3349,8 +3349,8 @@ void Hypo71::updateProfile(const string& name) {
 		setParameter("IMAG", imag);
 		config.readInto(kaz, "KAZ", blank);
 		setParameter("KAZ", kaz);
-		config.readInto(disableLastLoc, "DISABLE_LAST_LOC", blank);
-		setParameter("DISABLE_LAST_LOC", disableLastLoc);
+		config.readInto(useTrialPosition, "USE_TRIAL_POSITION", blank);
+		setParameter("USE_TRIAL_POSITION", useTrialPosition);
```
Comment (jmsaurel) [inline src/ipgp/plugins/locator/hypo71/hypo71.cpp]: Ok

File: src/ipgp/plugins/locator/hypo71/hypo71.cpp [hunk 4]
```diff
@@ -988,7 +988,7 @@ Origin* Hypo71::locate(PickList& pickList) {
 
 	// Available only on custom configuration file...
 	// Should we use the position obtained from the best ZTR value ?
-	if ( pConfig.read("USE_TRIAL_POSITION", true) ) {
+	if ( pConfig.read("USE_TRIAL_POSITION", false) ) {
```
Comment (jmsaurel) [inline src/ipgp/plugins/locator/hypo71/hypo71.cpp]: This one was already correct.
If `USE_TRIAL_POSITION = true` we want to use and start from the best trial run location, and we fix `cCC.[lat,lon]` to the `_trialLat and _trialLon` values.
If `USE_TRIAL_POSITION = false` we don't want to start from the best trial run location, and we let `cCC.[lat,lon]` to an empty value.

**Comment by jmsaurel** [comment]
The hypo71.cpp condition at line [991](https://github.com/SeisComP3/seiscomp3/blob/33cb572e236767538f69e5bf85907c6a684f27d8/src/ipgp/plugins/locator/hypo71/hypo71.cpp#L991) doesn't need to be updated.
It was already corrected when changing the variable name (but I missed the other occurrences you modified)

File: src/ipgp/plugins/locator/hypo71/hypo71.cpp (hunk 4 above)
Comment (gempa-jabe) [reply src/ipgp/plugins/locator/hypo71/hypo71.cpp]: The default.hypo71.conf set USE_TRIAL_POSITION to ```false```. Why do we read then USE_TRIAL_POSITION per default as ```true```? That means, if USE_TRIAL_POSITION is not given in any profile then its value is ```true```. Isn't that a mismatch?

Comment (jmsaurel) [reply src/ipgp/plugins/locator/hypo71/hypo71.cpp]: It seems I misunderstood `pConfig.read`, sorry.

`pConfig.read("USE_TRIAL_POSITION", false)`returns 
* false if `USE_TRIAL_POSITION = false` in a config file
* true if `USE_TRIAL_POSITION =  true` in a config file
* false if `USE_TRIAL_POSITION` is not found in any config file

Is that correct ?

Comment (gempa-jabe) [reply src/ipgp/plugins/locator/hypo71/hypo71.cpp]: Yes, that is correct. The value ```false``` is the fallback if the parameter has not been found.

Comment (gempa-jabe) [reply src/ipgp/plugins/locator/hypo71/hypo71.cpp]: Checking the code, it seems that ```USE_TRIAL_POSITION``` is not being read from the default control file. It is only used if defined in the control file of the actual profile. So it is not inherited. Just as a side note.

Comment (jmsaurel) [reply src/ipgp/plugins/locator/hypo71/hypo71.cpp]: Thanks for the clarification. Here is how it should behave (and how it is behaving with your code modification which is correct).

Default value of `USE_TRIAL_POSITION` is `false`
It's only an option, so not present in the default control file

**Comment by gempa-jabe** [comment]
If you think the PR is correct, I can merge it. If you are unsure about the default value in pConfig.read, I can remove it.

**Comment by jmsaurel** [comment]
PR is correct, default value should be false.

**Pull request merged**